test/e2e/multi-select.test.js

```diff
@@ -9,8 +10,129 @@ describe('vl-multi-select', async () => {
+        await assert.eventually.isFalse(multiselect.hasAttribute('changed'));
+        await multiselect.selectByValue('Germany');
+        await assert.eventually.isTrue(multiselect.hasAttribute('changed'));
+    });
+
+    it('Als gebruiker kan ik de multiselect via enable/disable-methode in/uit-schakelen', async () => {
+        const multiselect = await vlMultiSelectPage.getEnableDisableMethodeMultiselect();
+        await assert.eventually.isFalse(multiselect.isDisabled());
+        await vlMultiSelectPage.disable();
+        await assert.eventually.isTrue(multiselect.isDisabled());
+        await vlMultiSelectPage.enable();
+        await assert.eventually.isFalse(multiselect.isDisabled());
+    });
+
+    it('Als gebruiker kan ik programmatorisch (een) waarde(s) toevoegen/verwijderen in de multiselect', async () => {
+        const multiselect = await vlMultiSelectPage.getSetGetMethodeMultiselect();
+        await assert.eventually.isEmpty(multiselect.getSelectedOptionsByValue());
+        await vlMultiSelectPage.kiesDuitsland();
+        await assert.eventually.include(multiselect.getSelectedOptionsByValue(), 'Germany');
+        await vlMultiSelectPage.verwijderSelectie();
+        await assert.eventually.isEmpty(multiselect.getSelectedOptionsByValue());
+        await vlMultiSelectPage.kiesBelgieEnNederland();
+        await assert.eventually.include(multiselect.getSelectedOptionsByValue(), 'Belgium', 'Netherlands');
+    });
+
+    it('Als een multiselect boven een datepicker gerenderd wordt, kunnen zowel de multiselect als de datepicker correct gebruikt worden', async () => {
+        const multiselect = await vlMultiSelectPage.getDatepickerMultiselect();
+        const datepicker = await vlMultiSelectPage.getDatepicker();
+
+        const date = new Date();
+        const today = String(date.getDate()).padStart(1);
+        const dd = String(date.getDate()).padStart(2, '0');
+        const mm = String(date.getMonth() + 1).padStart(2, '0');
+        const yyyy = date.getFullYear();
+
+        await datepicker.selectDay(today);
+        await assert.eventually.equal(datepicker.getInputValue(), `${dd}.${mm}.${yyyy}`);
+
+        await multiselect.selectByValue('Germany');
+        await assert.eventually.include(multiselect.getSelectedOptionsByValue(), 'Germany');
+    });
+
+    it('Als gebruiker kan ik opties groeperen', async () => {
+        const multiselect = await vlMultiSelectPage.getGegroepeerdeMultiselect();
+        await assert.eventually.isTrue(multiselect.isGrouped());
+        await assert.eventually.isTrue(multiselect.hasHeadings());
+    });
+
+    it('De datepicker kan niet geopend worden wanneer de multiselect geopend is', async () => {
+        const multiselect = await vlMultiSelectPage.getDatepickerMultiselect();
+        const datepicker = await vlMultiSelectPage.getDatepicker();
+        await multiselect.openDropdown()
+        assert.isRejected(datepicker.selectDay(4));
```

Comment:
Is het niet beter om hier eerste de datepicker te openen, dan de multiselect en dan te checken of de datepicker nog zichtbaar is?

Reply:
Toegevoegd